accio-base/src/main/java/io/accio/base/client/duckdb/DuckDBDataSource.java

```diff
@@ -66,17 +70,25 @@ public Connection getConnection()
         Statement statement = connection.createStatement();
         statement.execute("set search_path = 'main'");
         // init httpfs settings
-        statement.execute(format("SET s3_endpoint='%s'", duckdbS3StyleStorageConfig.getEndpoint()));
-        statement.execute(format("SET s3_url_style='%s'", duckdbS3StyleStorageConfig.getUrlStyle()));
+        if (cacheStorageConfig instanceof DuckdbS3StyleStorageConfig) {
+            DuckdbS3StyleStorageConfig duckdbS3StyleStorageConfig = (DuckdbS3StyleStorageConfig) cacheStorageConfig;
+            statement.execute(format("SET s3_endpoint='%s'", duckdbS3StyleStorageConfig.getEndpoint()));
+            statement.execute(format("SET s3_url_style='%s'", duckdbS3StyleStorageConfig.getUrlStyle()));
+        }
+        if (duckDBSettingSQL.getSessionSQL() != null) {
+            statement.execute(duckDBSettingSQL.getSessionSQL());
+        }
```

Comment:
We can not expect the DuckDB is for what base on SQL is null. Because the SQL may be null at the begin for data source. If we want a basis, we can make DuckDBSettingSQL be null.

Reply:
Sure. If `DuckDBSettingSQL` is null, it's a DataSource. Otherwise, it's a PgMetastore.